spec-0012/index.md

```diff
@@ -4,122 +4,180 @@ number: 12
+regarded as `(...)`.
+- `x * y + z` is a compound expression; there are two operators and no explicit
+subexpressions that can be ignored.
+
+The acronym PEMDAS commonly refers to "parentheses", "exponential", "multiplication",
+"division", "addition", and "subtraction". Herein, we will consider these operators
+to be "PEMDAS operators", and we will also include the unary `+`, `-`, and `~` in
+this category for convenience. The order of operations of PEMDAS operators is typically
+taught in primary school and reinforced throughout a programmer's training and
+experience, so it is assumed that most programmers are comfortable relying on the
+implicit order of operations of expressions involving a few PEMDAS operations. Implicit
+order of operations becomes less obvious as the number of distinct operator priority
+levels increases and when multiple non-PEMDAS operators are involved. Portions of this
+acronym, namely MD and AS, will be used below to refer to the corresponding operators.
 
-## Notes
+## Implementation
 
-These formatting rules do not make any consideration in terms of performances
-nor precision. The scope is limited to styling.
+The following rules are not imagined to be all encompassing, and may be
+syntactically valid cases in which the following rules would lead to contradictions.
+Nonetheless, we expect them to be useful in most cases, and projects that adopt the
+SPEC endeavor to use them in typical cases. These rules are intended to respect and
+complement the PEP8 rules (relevant sections includes [id20](https://www.python.org/dev/peps/pep-0008/#id20) and [id20](https://www.python.org/dev/peps/pep-0008/#id28)).
+
+0. Unless required by other rules, rely on the implicit order of operations;
+   i.e., do not add extraneous parentheses. For example, prefer `u**v + y**z`
+   over `(u**v) + (y**z)`, and prefer `x + y + z` over `(x + y) + z`.
+1. Always use the `**` operator and unary `+`, `-`, and `~` operators *without*
+   surrounding whitespace. For example, prefer `-x**4` over `- (x ** 4)`. 
+2. Always surround non-PEMDAS operators with whitespace, and always make the priority of
+   non-PEMDAS operators explicit, even if the parentheses are consistent with the
+   implicit order of operations. For example, prefer `(x == y) or (w == t)` over
+   `x==y or w==t`.[^1]
+3. Always surround AS operators with whitespace.
+4. Typically surround MD operators with whitespace, except when there are lower-priority 
+   operators (namely AS) within the same compound expression or the division operation
+   would be written mathematically as a fraction with a horizontal bar.
+   For example, prefer `z = -x * y**t` over `z = -x*y**t`, but
+   prefer `z = w + x*y**t` over `z = w + x * y**t` due to the presence of the
+   lower-priority addition operator. Also, prefer `z = t/v * x/y` over
+   `z = t / v * x / y` if this would  be written mathematically as the product of
+   two fractions.
```

Comment:
I think this is the hairiest part of the whole thing - there are a lot of reasons why we might or might not want to include whitespace when it comes to `+-*/`. I expect this will need revision.